public/docs/ts/latest/guide/architecture.jade

```diff
@@ -1,190 +1,218 @@
+    source module is in the same folder (`./`) as the module importing it.
+    We could path up and around the application folder structure if the source module were somewhere else.
+  .l-sub-section
+    :marked
+      We import and export in the ECMAScript 2015 (ES2015) module syntax.
+      Learn more about that syntax [here](http://www.2ality.com/2014/09/es6-modules-final.html)
+      and many other places on the web.
+
+      The infrastructure *behind* module loading and importing is an important subject.
+      But it's a subject outside the scope of this introduction to Angular.
+      While we're focused on our application, *import* and *export*
+      is about all we need to know.
+
 :marked
   The key take-aways are:
   * Angular apps are composed of modules.
   * Modules export things &mdash; classes, function, values &mdash; that other modules import.
   * We prefer to write our application as a collection of modules, each module exporting one thing.
-  
+
   The first module we write will most likely export a component.
+
 .l-main-section
-<a id="component"></a>
 :marked
-  ## The Component
+  ## Components
 figure
   img(src="/resources/images/devguide/architecture/hero-component.png" alt="Component" align="left" style="width:200px; margin-left:-40px;margin-right:10px" )
+
 :marked
-  A **Component** controls a patch of screen real estate that we could call a *view*. 
-  The shell at the application root with navigation links, that list of heroes, the hero editor ... 
-  they're all views controlled by Components.
-  
-  We define a Component's application logic - what it does to support the view - inside a class.
-  The class interacts with the view through an API of properties and methods. 
-  
+  A **component** controls a patch of screen real estate that we could call a *view*.
+  The shell at the application root with navigation links, a list of heroes, a hero editor ...
+  they're all views controlled by components.
+
+  We define a component's application logic &mdash; what it does to support the view &mdash; inside a class.
+  The class interacts with the view through an API of properties and methods.
+
   <a id="component-code"></a>
-  A `HeroListComponent`, for example, might have a `heroes` property that returns an array of heroes
+  A `HeroListComponent`, for example, might have a `heroes` property that returns an !{_array} of heroes
   that it acquired from a service.
-  It might have a `selectHero()` method that sets a `selectedHero` property when the user clicks on a hero from that list.
-  It might be a class like this:
+  It might have a `selectHero()` method that sets a `selectedHero` property when the user clicks to choose a hero from that list.
+  The component might be a class like this:
 
-+makeExample('architecture/ts/app/hero-list.component.ts', 'class', 'app/hero-list.component.ts')
++makeExcerpt('app/hero-list.component.ts', 'class')
```

Comment:
Lifecycle hooks chapter is mentioned in the paragraph that follows. I've added an explicit mention of `ngOnInit`.